plugins/woocommerce/src/Internal/Logging/RemoteLogger.php

```diff
@@ -16,10 +18,183 @@
  * @since 9.2.0
  * @package WooCommerce\Classes
  */
-class RemoteLogger {
+class RemoteLogger extends \WC_Log_Handler {
+	const LOG_ENDPOINT                = 'https://public-api.wordpress.com/rest/v1.1/logstash';
+	const RATE_LIMIT_ID               = 'woocommerce_remote_logging';
+	const RATE_LIMIT_DELAY            = 60; // 1 minute.
 	const WC_LATEST_VERSION_TRANSIENT = 'latest_woocommerce_version';
 	const FETCH_LATEST_VERSION_RETRY  = 'fetch_latest_woocommerce_version_retry';
 
+	/**
+	 * The logger instance.
+	 *
+	 * @var \WC_Logger_Interface|null
+	 */
+	private $local_logger;
+
+	/**
+	 * Remote logger constructor.
+	 *
+	 * @internal
+	 * @param \WC_Logger_Interface|null $logger Logger instance.
+	 */
+	public function __construct( \WC_Logger_Interface $logger = null ) {
+		if ( null === $logger ) {
+			$this->local_logger = wc_get_logger();
+		}
+		$this->local_logger = $logger;
+	}
+
+	/**
+	 * Handle a log entry.
+	 *
+	 * @param int    $timestamp Log timestamp.
+	 * @param string $level emergency|alert|critical|error|warning|notice|info|debug.
+	 * @param string $message Log message.
+	 * @param array  $context Additional information for log handlers.
+	 *
+	 * @throws \Exception If the remote logging fails. The error is caught and logged locally.
+	 *
+	 * @return bool False if value was not handled and true if value was handled.
+	 */
+	public function handle( $timestamp, $level, $message, $context ) {
+		if ( ! \WC_Log_Levels::is_valid_level( $level ) ) {
+			/* translators: 1: WC_Remote_Logger::log 2: level */
+			wc_doing_it_wrong( __METHOD__, sprintf( __( '%1$s was called with an invalid level "%2$s".', 'woocommerce' ), '<code>WC_Remote_Logger::handle</code>', $level ), '9.2.0' );
+		}
+
+		if ( ! $this->is_remote_logging_allowed() ) {
+			return false;
+		}
+
+		if ( $this->is_third_party_error( (string) $message, (array) $context ) ) {
+			return false;
+		}
+
+		if ( WC_Rate_Limiter::retried_too_soon( self::RATE_LIMIT_ID ) ) {
+			$this->local_logger->info( 'Remote logging throttled.', array( 'source' => 'wc-remote-logger' ) );
+			return false;
+		}
+
+		try {
+			$log_data = $this->get_formatted_log( $level, $message, $context );
+
+			// Ensure the log data is valid.
+			if ( ! is_array( $log_data ) || empty( $log_data['message'] ) || empty( $log_data['feature'] ) ) {
+				return false;
+			}
+
+			$body = array(
+				'params' => wp_json_encode( $log_data ),
+			);
+
+			\WC_Rate_Limiter::set_rate_limit( self::RATE_LIMIT_ID, self::RATE_LIMIT_DELAY );
```

Comment:
Minor and non-essential change (so feel free to skip):

```suggestion
			WC_Rate_Limiter::set_rate_limit( self::RATE_LIMIT_ID, self::RATE_LIMIT_DELAY );
```

(We import `WC_Rate_Limiter`, so qualification isn't needed.)